### ISSUE TITLE
JAWS doesn't seem to read "Looking for a job" and "Looking to hire" features in a way that is equitable to a visual experience

### ISSUE THREAD

**Issue by RM-1978** [GCTC-NTGC/gc-digital-talent#5376]
## ❌ Current behaviour (issue)
JAWS only reads "Looking for a job" and "Looking to hire" buttons/links only when the link list is called up. It does not pick them up when a user browses the page. The tester suggested it was because of how JAWS handles buttons and links differently. 
### Location: 
Home page 
### AT configuration:  
JAWS, Windows 11, Edge
### Category: 
Accessibility Bug 
### How to reproduce

1. Access the Home page with JAWS.
2. Tab through past the main navigation utility to test if the focus picks up the “Looking for a job” and “Looking to hire” buttons in the Tab order.
3. Listen for these buttons to be read by a screen reader without using the link list JAWS feature.
## ✔ Expected behaviour (fix)
Screen reader reads the “ Looking for a job” and “Looking to hire” buttons.
### Acceptance Criteria
- [ ] JAWS reads these features as Buttons. 
## Additional information (optional)
How does it impact users? 
1. It impacts a candidate screen reader user's option to jump straight to either the job opportunities page.
2. It impacts a manager screen reader user's option to jump  straight to the search page.

**Comment by substrae** [comment]
@esizer Any insight on this? I don't believe we're doing anything special to the anchor element in this case. They appear properly in the tab order, so this isn't tab-order related, rather, it likely has to do with JAWS specifically misreading the markup. We might need more context or video to debug.

**Comment by RM-1978** [comment]
https://app.makeitfable.com/clip/63c74d8ce50341002dc3b466

**Comment by esizer** [comment]
> @esizer Any insight on this? I don't believe we're doing anything special to the anchor element in this case. They appear properly in the tab order, so this isn't tab-order related, rather, it likely has to do with JAWS specifically misreading the markup. We might need more context or video to debug.

Odd, just tested and cannot reproduce.

- Windows 11 22621.1105
- Edge 109.0.1518.55 
- JAWS 2023.2212.23

**Comment by RM-1978** [comment]
Thanks Eric!  I think it would be good to take a look at the Fable Clip to see how the user produced the error: [Looking for a Job/Looking to hire links with JAWS](https://app.makeitfable.com/clip/63c74d8ce50341002dc3b466)

Once you access the link, you can get a Fable Access code sent to your email address  so that you can view the video of the situation.

**Comment by esizer** [comment]
> Thanks @esizer! I think it would be good to take a look at the Fable Clip to see how the user produced the error: [Looking for a Job/Looking to hire links with JAWS](https://app.makeitfable.com/clip/63c74d8ce50341002dc3b466)

Thanks, I did not see that at first.

After watching the video this may be a terminology issue. They are actually anchors styled to be buttons so we do not expect them to appear in the list of buttons with JAWS which makes a distinction between the two.

**Comment by substrae** [comment]
Ah - yes - it's important to distinguish between what is _functionally_ a button vs. what visually _appears_ to be a button. Unfortunately, the functional difference between buttons and anchors doesn't take the visual differences into account; a button triggers a script, while an anchor links to a location. This doesn't mean, however, that all objects that link to a location will always look like an anchor, because the affordance provided by a button is important in helping users understand where to click.

**Comment by petertgiles** [comment]
So to confirm, this was a problem with the test?  Or is there something for devs to do here?

**Comment by petertgiles** [comment]
So the requested change is to turn them from links to buttons?  Or make them look like links?  Sorry, I'm having trouble pinning down the solution.

**Comment by mnigh** [comment]
This might be resolved with #6380, #6358.

**Comment by petertgiles** [comment]
We haven't been able to come up with an action plan for this one.  Closing for now but we can reopen if we come up with a solution.

**Comment by RM-1978** [comment]
I am thinking Ice Box again

**Comment by esizer** [comment]
> Hum, @mnigh if it is an instruction or label issue, do we just leave it or do we update the copy/ARIA label everywhere? In the latter case, the updates would for Copy caucus might be:
> 
> 1. `Activate this button to...` or
> 2. `Activate this link to access...`

I believe the issue is with how we prompted the user to complete the task, rather than an issue with the result of the test. Based on the task we asked the user to complete, the video shows the expected end result since there are no buttons on the home page unless you are logged in (then "logout" is a button).

It is not necessarily the instructions or label within the links themselves, rather the instructions/prompt we gave the user during the fable test.

In the video, we prompted the user to "find the _button_ that would allow you to hire talent". In response to the prompt, the user used the [list buttons](https://www.freedomscientific.com/training/jaws/hotkeys/#:~:text=List%20Buttons,CTRL%2BINSERT%2BB) function of JAWS to find the button. There are no buttons on the page that allow you to do this so, we don't expect anything useful to show up in that list.

If we were to re-run the same test and change the prompt to "find the _link_ that would allow you to hire talent" I would expect to see a different more positive outcome since the user may use the [list links](https://www.freedomscientific.com/training/jaws/hotkeys/#:~:text=List%20Links,INSERT%2BF7) function in JAWS instead which would show all links on the page, including the two mentioned in this ticket.

**Comment by esizer** [comment]
I can confirm, the links do appear in the focus order and are appearing in the list of links 👍 


https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/2397f1f7-6e3e-400a-b9e0-fdf154a52727


https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/6710e5a7-65ac-43c9-a435-941d91678f79

**Comment by RM-1978** [comment]
@petertgiles  Why are we closing this as not planned? . @substrae had it right. It's important to distinguish between what is functionally a button vs. what visually appears to be a button. We need to ensure that our instructions  or labels, do we leave it or do we update the copy/ARIA label everywhere?

Because this is closed,  I assume that we have chosen the latter case, the updates would for the copy work that Dasha and I are doing?

Would you now like updates for  insitructions for instructions to be: 

- Activate this button to... or
- Activate this link to access...

I will leave the ARIA labels to you.

**Comment by petertgiles** [comment]
Hi Lee.  I closed the issue because it seemed that the consensus was that AT was correctly finding the links on this page and no further dev actions were needed.  If I misinterpreted feel free to reopen and we'll refocus.  It's my opinion that if the next action is to update some ARIA labels it would be better to start a new issue with the specific new labels and which elements to update.

**Comment by RM-1978** [comment]
Sounds good re the new issue. @mnigh do you think that adjuting the ARIA Labels and Instructions for the correct language  might be a big enough issue to bring to copy caucus? 

If the ATs are distinguishing between buttons and links, then the issue is definitely ok to be closed.

**Comment by substrae** [comment]
@RM-1978 @petertgiles 

I'm a little confused by the discussion here. As I understand it, because we asked the user to find a "button" using the AT during the test, they couldn't find the intended action on the page because the action was in fact a functional link (`<a>`), not a `<button>` element. 

If this is the case, this was a problem with the test question, not the app, because as the testers, we know the functional difference between links and buttons, so we should have anticipated this task being impossible to complete (we know that asking them to navigate through buttons will never turn up the correct result). Moving forward, if we're testing to find an action, we should be asking "Find the button or link..." to ensure that users understand they need to browse across both. The tester shouldn't be responsible for understanding whether we're asking them to find an element that links somewhere or triggers an action.

@esizer Correct me if I'm wrong, but updating the `aria-label` in this instance won't actually make a difference, because if we repeated the test with the original question, the label wouldn't tell the AT that a link should be treated as a button and vice-versa.

**Comment by esizer** [comment]
> @esizer Correct me if I'm wrong, but updating the `aria-label` in this instance won't actually make a difference, because if we repeated the test with the original question, the label wouldn't tell the AT that a link should be treated as a button and vice-versa.

I think maybe if we are doing another test we attempt this same prompt again but, as you said, specify "link or button" instead of just "button". I believe this will lead to a successful test and if not, then we can look at either changing the rendered text or implement an `aria-label`. 

We already changed them from question based to action based which should help as well. 👍

**Comment by RM-1978** [comment]
So @substrae, the test was really to accomplish certain tasks using our platform. Going forward, I will be less specific about how the accomplish tasks, and see id the Fable Testers have any more trouble with button and link interactions.
